src/typing_extensions.py

```diff
@@ -1468,8 +1468,9 @@ def __reduce__(self):
 def _set_default(type_param, default):
     type_param.has_default = lambda: default is not NoDefault
     if isinstance(default, (tuple, list)):
-        type_param.__default__ = tuple((typing._type_check(d, "Default must be a type")
-                                        for d in default))
+        type_param.__default__ = type(default)(
+            typing._type_check(d, "Default must be a type") for d in default
+        )
```

Comment:
The `_type_check()` calls here feel "morally correct" -- they're consistent with what a lot of the rest of the typing.py internals do. But on the other hand, CPython doesn't actually do any check at all in its 3.13 implementation; it just sets the `__default__` attribute to the value passed in, and moves on.

Reply:
So maybe we should just do that here too?

Reply:
I wasn't sure whether the bug was that we were being too lenient or that CPython wasn't being restrictive enough. But having slept on it, I now agree that we should just accept any default value, and not do any type checking here. That's consistent with what CPython has always done for TypeVar bounds and TypeVar constraints, which should also conceptually be types:

```pycon
% pyenv local 3.11                                                                                                % python
Python 3.11.8 (main, Feb 15 2024, 19:32:18) [Clang 15.0.0 (clang-1500.1.0.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from typing import TypeVar
>>> def not_a_type(): pass
... 
>>> T = TypeVar("T", bound=not_a_type)
>>> U = TypeVar("U", not_a_type, not_a_type)
```